### PR TITLE
[Catalog] Cover Component/FilterFactory by Unit Test

### PR DESCRIPTION
**Pull request by edenduong**
### Description (*)

**Increase Test Coverage**

[Catalog] Cover Component/FilterFactory by Unit Test

Cover Class `Magento\Catalog\Ui\Component\FilterFactory`

### Manual testing scenarios (*)

**vendor/bin/phpunit app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php**

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-6468](https://jira.corp.magento.com/browse/ENGCOM-6468) has been created to process this Pull Request

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+class FilterFactoryTest extends TestCase
+{
+    /**
+     * @var FilterFactory
+     */
+    private $filterFactory;
+
+    /**
+     * @var UiComponentFactory|\PHPUnit_Framework_MockObject_MockObject
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: Alias `\PHPUnit_Framework_MockObject_MockObject` as `MockObject`

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Catalog\Test\Unit\Ui\Component;
+
+use PHPUnit\Framework\TestCase;
+use Magento\Catalog\Ui\Component\FilterFactory;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager as ObjectManagerHelper;
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: You don't really need to suffix with `Helper`. `ObjectManager` is quite clear.

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            [
+                [
+                    'value' => 1,
+                    'label' => 'Black',
+                ],
+                [
+                    'value' => 2,
+                    'label' => 'White',
+                ]
+            ]
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: You can extract that to private const `STUB_COLOR_OPTIONS` for example.

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+                    'config' => [
+                        'options' => [
+                            [
+                                'value' => 1,
+                                'label' => 'Black',
+                            ],
+                            [
+                                'value' => 2,
+                                'label' => 'White',
+                            ]
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: Then refer to that STUB.

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    /**
+     * Test create() with use source attribute
+     */
+    public function testCreateWithUseSourceAttribute()
+    {
+        $contextMock = $this->createMock(ContextInterface::class);
+        $attributeMock = $this->getMockBuilder(ProductAttributeInterface::class)
+            ->setMethods(['usesSource', 'getSource'])
+            ->getMockForAbstractClass();
+        $attributeMock->method('getAttributeCode')->willReturn('color');
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: Also - extract that to private const `STUB_ATTRIBUTE_CODE`

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                                'value' => 1,
+                                'label' => 'Black',
+                            ],
+                            [
+                                'value' => 2,
+                                'label' => 'White',
+                            ]
+                        ],
+                        'caption' => (string)__('Select...'),
+                        'dataScope' => 'color',
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: Refer to `STUB_ATTRIBUTE_CODE`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
You may also `Optimize imports` in your PHPStorm to sort the imports with alphabetic order.

**Comment by edenduong** [comment]
Hi @lbajsarowicz: Thank you. I have changed the source code. Please check it.

File: app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            ->willReturn(self::STUB_ATTRIBUTE['default_frontend_label']);
+        $attributeMock->method('usesSource')->willReturn(self::STUB_ATTRIBUTE['uses_source']);
+        $attributeMock->method('getSourceModel')->willReturn(self::STUB_ATTRIBUTE['source_model']);
+        $attributeMock->method('getFrontendInput')->willReturn(self::STUB_ATTRIBUTE['frontend_input']);
+        $sourceMock = $this->createMock(SourceInterface::class);
+        $attributeMock->method('getSource')->willReturn($sourceMock);
+        $sourceMock->method('getAllOptions')->willReturn(self::STUB_ATTRIBUTE['all_options']);
+        $this->componentFactoryMock->expects($this->once())
+            ->method('create')
+            ->with('color', 'filterSelect', [
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Ui/Component/FilterFactoryTest.php]: `self::STUB_ATTRIBUTE['attribute_code']`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
I haven't expected that kind of stub, but it's also fine.
Last replacement and I'll accept the PR.

**Comment by edenduong** [comment]
I have changed it. Please check it again. Thanks @lbajsarowicz

**Comment by edenduong** [comment]
Could you please take a look? Thanks @lbajsarowicz

**Review by lbajsarowicz — APPROVED** [review]
Thank you for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6468](https://jira.corp.magento.com/browse/ENGCOM-6468) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

**Pull request merged**